compiler/rustc_hir_analysis/src/check/check.rs

```diff
@@ -722,7 +722,9 @@ pub(super) fn check_specialization_validity<'tcx>(
     let result = opt_result.unwrap_or(Ok(()));
 
     if let Err(parent_impl) = result {
-        report_forbidden_specialization(tcx, impl_item, parent_impl);
+        if !tcx.is_impl_trait_in_trait(impl_item) {
```

Comment:
I've added this to avoid calling `report_forbidden_specialization` because all it does https://github.com/spastorino/rust/blob/518f84ba766d503cb709f753a0a498d87b942dd0/compiler/rustc_hir_analysis/src/check/mod.rs#L172-L182 is emit an error about an impl element that is not marked as default and that doesn't make any sense for a synthesized RPIT.
Do we want a comment in the code about this?